plugins/inputs/t128_metrics/README.md

```diff
@@ -13,6 +13,10 @@ The metrics input plugin collects metrics from a 128T instance.
 ## A socket to use for retrieving metrics - unused by default
 # unix_socket = "/var/run/128technology/web-server.sock"
 
+## Whether or not to use the bulk retrieval API. If used, the base_url should
+## point directly to the bulk metrics endpoint.
+# use_bulk_retrieval = false
```

Comment:
I'm assuming the default has to be `false` for backward compatibility, correct? Having said that, when available is there a reason to not let this be the default mode? Said differently, can we just make the collector smart enough where it can check whether the bulk retrieval endpoint exists on the local MARS instance or not? For pre 5.2 routers, this check will be "false" and for 5.2 and beyond it will be true.

I'm not opposed to having the flag as a backdoor to override this default detection but I think it would be nice to just handle it at the collector level and not have to manage different version of 128T software. 

This does not have to be done at the collector level, just one time by the agent on upgrades and what not. The result can be cached and the collector just needs to know where to look.  This will allow for easily upgrading existing deployments to use the better functionality.  

Thoughts?

Reply:
You're correct. For >=5.2, there won't be any reason not to use this field. However, I want to keep the collector more or less oblivious to 128T versions etc. So this `use_bulk_retrieval=true` will be automatically injected by the Monitoring Agent during translation.